src/hotspot/share/runtime/threads.hpp

```diff
@@ -102,6 +102,15 @@ class Threads: AllStatic {
   static void change_thread_claim_token();
   static void assert_all_threads_claimed() NOT_DEBUG_RETURN;
 
+  struct ThreadsClaimTokenScope : StackObj {
+    ThreadsClaimTokenScope() {
+      Threads::change_thread_claim_token();
+    }
+    ~ThreadsClaimTokenScope() {
+      Threads::assert_all_threads_claimed();
+    }
+  };
```

Comment:
I wonder if the implementation should be moved to the cpp file or if we consider this small enough to include here?